core/storage/statistics/gae_models.py

```diff
@@ -1134,6 +1138,128 @@ def delete_playthroughs_multi(cls, playthrough_ids):
         cls.delete_multi(instances)
 
 
+class LearnerAnswerDetailsModel(base_models.BaseModel):
+    """Model for storing the answer details that a learner enters when they
+    are asked for an explanation of their answer. Currently, the model supports
+    exploration states and questions. One instance of this model is created for
+    each of these objects.
+    The id of this model will be the hash which is automatically generated when
+    the model instance is created.
+    """
+
+    # The reference to the state for which model instance is created.
+    # For exploration state the state reference is of the form
+    # 'exp_id'.'state_name', while for question state reference is of the form
+    # 'question_id' as currently the one question holds only one state.
+    state_reference = ndb.StringProperty(required=True, indexed=True)
+    # The type of entity e.g "exploration" or "question".
+    entity_type = ndb.StringProperty(
+        required=True, indexed=True, choices=ALLOWED_ENTITY_TYPES)
+    # The id of the interaction for which the answer details were received.
+    interaction_id = ndb.StringProperty(required=True, indexed=True)
+    # List of LearnerAnswerInfo dicts, which is defined in
+    # stats_domain.py, each dict corresponds to a single answer info of
+    # learner.
+    learner_answer_info_list = ndb.JsonProperty(repeated=True, indexed=False)
+    # The version of the learner_answer_info_list currently supported by Oppia.
+    # If the internal JSON structure of learner_answer_info_list changes,
+    # CURRENT_SCHEMA_VERSION in this class needs to be incremented.
+    schema_version = ndb.IntegerProperty(
```

Comment:
Give this a longer name that reflects what it is the schema version of.

Reply:
Done